filebeat/input/syslog/syslog_rfc3164.rl

```diff
@@ -0,0 +1,45 @@
+%%{
```

Comment:
Interesting!

Reply:
I really like this approach. Do you mind sharing why did you choose ragel? What other lexers did you consider? I am just curious.

Reply:
I haven't look too much in other lexer other than https://github.com/timtadh/lexmachine

I've worked with ragel before, I've made a SQL like DSL on top of a lucene document. They really focus on speed and this is a mature project.

Also, I like the fact that the lexer and the target language are not linked, Meaning we can target both the beats syslog input and the logstash input syslog. But ragel isn't the only tool available that can do that.